crowbar_framework/lib/crowbar/upgrade.rb

```diff
@@ -100,6 +100,17 @@ def knife_files
           filecontent_replace(file, "nova_dashboard", "horizon")
         end
       end
+
+      # find admin node and update target_platform
+      nodes_path = knife_path.join("nodes")
+      nodes_path.children.each do |file|
+        json = JSON.load(file.read)
+        next unless json["crowbar"] && json["crowbar"]["admin_node"]
+        json["target_platform"] = "suse-12.1"
```

Comment:
Yeah, I wondered about that as well. Couldn't come up with a good idea yet. Do we have any (non-clumsy) way to map from the current information in /etc/os-release to a platform string that chef likes? I guess ohai provides that, but it's probably a bit too much to run ohai here?